lib/core/src/recover/handlers/handle_receive_swap.rs

```diff
@@ -45,14 +45,23 @@ impl ReceiveSwapHandler {
         let swap_id = &receive_swap.id.clone();
         debug!("[Recover Receive] Recovering data for swap {swap_id}");
 
-        let mrh_script = ElementsAddress::from_str(&receive_swap.mrh_address)
-            .map_err(|_| anyhow::anyhow!("Invalid MRH address for swap {swap_id}"))?
-            .script_pubkey();
+        let mrh_script = if !receive_swap.mrh_address.is_empty() {
+            match ElementsAddress::from_str(&receive_swap.mrh_address) {
+                Ok(addr) => Some(addr.script_pubkey()),
+                Err(_) => {
+                    warn!("Invalid MRH address format for swap {swap_id}. Skipping MRH recovery.");
```

Comment:
Might as well log the error
```suggestion
                Err(e) => {
                    warn!("Invalid MRH address format for swap {swap_id} ({e:?}). Skipping MRH recovery.");
```